cmd2/history.py

```diff
@@ -7,25 +7,29 @@
 
 from typing import List, Union
 
+import attr
+
 from . import utils
 from .parsing import Statement
 
 
-class HistoryItem(str):
+@attr.s(frozen=True)
+class HistoryItem():
     """Class used to represent one command in the History list"""
-    listformat = ' {:>4}  {}\n'
-    ex_listformat = ' {:>4}x {}\n'
+    _listformat = ' {:>4}  {}\n'
+    _ex_listformat = ' {:>4}x {}\n'
 
-    def __new__(cls, statement: Statement):
-        """Create a new instance of HistoryItem
+    statement = attr.ib(default=None, validator=attr.validators.instance_of(Statement))
+    idx = attr.ib(default=None, validator=attr.validators.instance_of(int))
 
-        We must override __new__ because we are subclassing `str` which is
-        immutable and takes a different number of arguments as Statement.
-        """
-        hi = super().__new__(cls, statement.raw)
-        hi.statement = statement
-        hi.idx = None
-        return hi
+    def __str__(self):
+        """A convenient human readable representation of the history item"""
+        return self.statement.raw
```

Comment:
@kotfu I simplified the implementation for this since the attrs validator assures we have a Statement.  Please make sure this and the unit test I added seems OK to you.

Reply:
Looks good. I'm working to get `history.py` and `do_history()` to 100% test coverage before we merge this.

Reply:
This should put `history.py` at 100% coverage.  But there are a number of little edge cases in `do_history` remaining.  Good luck!